File: split/xovern.pas
function xovern(x: scaled; n: integer): scaled;
var
  negative: boolean;
begin
  negative := false;
  if n = 0 then
  begin
    aritherror := true;
    xovern := 0;
    remainder := x;
  end
  else
  begin
    if n < 0 then
    begin
      x := -x;
      n := -n;
      negative := true;
    end;
    if x >= 0 then
    begin
      xovern := x div n;
      remainder := x mod n;
    end
    else
    begin
      xovern := -((-x) div n);
      remainder := -((-x) mod n);
    end;
  end;
  if negative then
    remainder := -remainder;
end;

File: cpp/src/xovern.cpp
#include "xovern.h"

scaled xovern(scaled x, int n)
{
	bool negative = false;
	if (n == 0)
	{
		aritherror = true;
		remainder_ = x;
		return 0;
	}
	if (n < 0)
	{
		x = -x;
		n = -n;
		negative = true;
	}
	if (x >= 0)
	{
		remainder_ = negative ? -x%n : x%n;
		return x/n;
	}
	remainder_ = negative ? (-x)%n : -((-x)%n);
	return -((-x)/n);
}

